File: Python Excel Examples/WorksheetsApiDemo/updateWorksheetProperty.py
import spirecloudexcel
from spirecloudexcel.configuration import Configuration as ExcelConfiguration
from spirecloudexcel.models import Color

appId = "your id"
appKey = "your key"
baseUrl="https://api.e-iceblue.cn"
configuration = ExcelConfiguration(appId, appKey,baseUrl)
api = spirecloudexcel.api.worksheets_api.WorksheetsApi(configuration)

name = "UpdateWorksheetProperty.xlsx"
folder = "ExcelDocument";
storage = ""
sheetName = "Sheet1"
sheet = spirecloudexcel.models.Worksheet()
sheet.zoom = 50;
sheet.index = 3
sheet.visibility_type = "hidden"
sheet.is_gridlines_visible = False
sheet.tab_color = Color(255, 255, 0, 0)  # 设置选项卡颜色
sheet.is_protected = True
sheet.is_selected = True
api.update_worksheet_property(name, sheetName, sheet, folder=folder, storage=storage)

File: JavaScript Excel Examples/WorksheetsApiDemo/updateWorksheetProperty.js
(function (){
    var appId = "your id";
    var appKey = "your key"; 
    
    var baseUrl = "https://api.e-iceblue.cn";

    var SpirecloudExcel = require('../../src/index');
    var apiClient = new SpirecloudExcel.ApiClient(appId, appKey,baseUrl);
    var instance = new SpirecloudExcel.WorksheetsApi(apiClient);

    var name = "UpdateWorksheetProperty.xlsx";
    var opts = {
        folder: "input"
    };
    var sheetName = "Sheet2";
    var sheet = new SpirecloudExcel.Worksheet();
    sheet.Zoom = 50;
    sheet.Index = 3;
    sheet.IsGridlinesVisible = false; 
    sheet.IsSelected = true;
    sheet.IsProtected = true;
    sheet.TabColor = new SpirecloudExcel.Color(); 
    sheet.TabColor.A = 255;
    sheet.TabColor.R = 255;
    sheet.TabColor.G = 0;
    sheet.TabColor.B = 0;
    instance.updateWorksheetProperty(name, sheetName, sheet, opts, function (error, data, response) {
            if (error) throw error;
        });
})();